tools/nodejs_api/src_nodejs/connection.js

```diff
@@ -4,17 +4,44 @@ const QueryResult = require("./queryResult.js");
 class Connection {
     #connection;
     #database;
+    #initPromise;
+    #initialized;
     constructor(database, numThreads = 0) {
-        if (typeof database !== "object" || database.constructor.name !==  "Database" || typeof numThreads !== "number" || !Number.isInteger(numThreads)){
+        if (typeof database !== "object" || database.constructor.name !== "Database" || typeof numThreads !== "number" || !Number.isInteger(numThreads)) {
             throw new Error("Connection constructor requires a database object and optional numThreads integer as argument(s)");
         }
         this.#database = database;
         this.#connection = new kuzu.NodeConnection(database.database, numThreads);
+        this.#initPromise = this.#connection.getConnection();
```

Comment:
You should not start initializing upon calling constructor. It will make it hard for the user to catch the initialization error. Do the initialization upon the user issues the first query.